cpp/oneapi/dal/backend/common.hpp

```diff
@@ -71,11 +71,16 @@ template <typename Integer>
 inline constexpr Integer down_pow2(Integer x) {
     static_assert(std::is_integral_v<Integer>);
     ONEDAL_ASSERT(x > 0);
-    Integer power = 1;
-    while (power < x / 2) {
-        power *= 2;
+    Integer power = 0;
+    if (!(x & (x - 1))) {
```

Comment:
Why you handle cases `x==0` and `x==1` when this function has an assert at line 73 that prohibits calling it with `x==0`?

Reply:
This is check for power of two, if `x` is already power of 2, do nothing 😅

Reply:
Suggest to have separate function for that, maybe someone needs it

Reply:
```c++
inline constexpr bool(...)
```

Reply:
Added check for power of two